examples/pic.cpp

```diff
@@ -0,0 +1,363 @@
+            pairCh.push(p, i*totDataPoints + p.id() );
+        }
+        for (uint i = p.id()+1; i < totDataPoints; ++i) {
+            pairCh.push(p, p.id()*totDataPoints + i );
+        }
+    });
+
+    husky::base::log_msg( "affMat size " + std::to_string(affMatElem_list.get_size()));
+
+    husky::ObjList<DimStat> rowstat_list;
+    auto& rowsumCh =
+        husky::ChannelFactory::create_push_combined_channel<float, husky::SumCombiner<float>>(affMatElem_list, rowstat_list);
+
+    list_execute(affMatElem_list, [&](MatElem & e){
+        auto pts = pairCh.get(e);
+        float nSigmaSq = 0;
+
+        //if(e.id()==10011){
+        //    husky::base::log_msg( "========" );
+        //    for (auto aVal : pts){ husky::base::log_msg( "getval " + std::to_string(aVal.coords[0]));}
+        //    husky::base::log_msg( "========" + std::to_string(local_dimstat_list[0]->var) );
+        //}
+
+        for (uint i =0; i<dimension;i++){ nSigmaSq += pow( pts[0].coords[i] - pts[1].coords[i], 2)/ (2.*dimstatBCh.get(i).var); }
+        e.val = exp( -nSigmaSq);
+        int row = e.id()/totDataPoints;
+        int col = e.id()%totDataPoints;
+        rowsumCh.push( e.val, row);
+        rowsumCh.push( e.val, col);
+        rowsumCh.push( e.val, -1 ); //hack, -1 is for calculating the volume (sum of all elements) of the affMat
+    });
+
+    //calculate then boardcast the rowsum. (is there a push_combine_boardcast channel API?)
+    auto& rowsumBCh = husky::ChannelFactory::create_broadcast_channel<int, float>(rowstat_list);
+    list_execute(rowstat_list, [&](DimStat & r){
+        rowsumBCh.broadcast( r.id(), rowsumCh.get(r) );
+    });
+
+    husky::base::log_msg( "part3 ok ");
+
+    // 4. repeat multiply the affinity matrix to the vector till convergence
+
+    //To row normalized the affMat we need to calculate D*A, where D is diagonal matrix with 1/rowsum at the diagonal
+    //We want Vn = (D*A) * ... * (D*A) * (D*A) * V0
+    //Instead of calculating D*A, we can just multiply D on the A*V0 vector
+    //ie. V1 = D*(A*V0)
+    //    V2 = D*(A*V1)...
+
+    float affMatVol = rowsumBCh.get(-1)*2.;
+
+    husky::ObjList<VElem> velem_list;
+    int chunkSize = totDataPoints/husky::Context::get_worker_info()->get_num_workers()+1;
+    int startPos  = chunkSize * husky::Context::get_global_tid();
+    int endPos    = std::min( int(totDataPoints) , chunkSize * (husky::Context::get_global_tid()+1));
+
+    //Init V0
+    for (uint i = startPos; i<endPos; i++){
+        VElem e(i);
+        e.setVal( rowsumBCh.get(i) / affMatVol );
+        velem_list.add_object( e );
+    }
+    globalize(velem_list);
+
+    husky::base::log_msg( "vec "
+                          + std::to_string(velem_list.get_size()) + " "
+                          + std::to_string(startPos) + " "
+                          + std::to_string(endPos) + " "
+                        );
+
+    //FIXME:: Instead of the whole object, should push only the val and id of VElem to the Mat
```

Comment:
Usually it's `FIXME(name)`, in which the `name` refers to person who should handle it. @kygx-legend